include/cutlass/gemm/collective/builders/xe_mma_builder.inl

```diff
@@ -38,8 +38,154 @@
+      auto atoms_in_m = tile_M / atom_m;
+      for (auto atoms : {8,4,2}) {
+        if (atoms_in_m >= atoms) {
+          return atoms;
+        }
+      }
+    }
+    return 1;
+  } else {
+    // Transpose loads are always size 16 in the M dim (for bf16).
+    static_assert(tile_M / 16 > 0 and tile_M % 16 == 0, "Invalid Tile size in M dim");
+    return tile_M / 16;
+  }
+}
+
+// Generic way to pick a copy atom for A
+// If tuning for a specific case, create a SubgroupTilingMap specialization
+template <typename LayoutA, class TileShape_MNK, int sgs_M>
+inline auto pick_load_atom_for_A() {
+  if constexpr (cute::is_same_v<LayoutA, cutlass::layout::RowMajor>) {
+    constexpr int tile_M = get<0>(TileShape_MNK{});
+    constexpr int tile_K = get<2>(TileShape_MNK{});
+    static_assert(tile_M % sgs_M == 0);
+    constexpr int atoms_in_M_dim = tile_M / sgs_M;
+    if constexpr (atoms_in_M_dim >= 32 && tile_K >= 32) {
+      return XE_2D_U16x32x32_LD_N{};
+    } else if constexpr (atoms_in_M_dim >= 32) {
+      return XE_2D_U16x32x16_LD_N{};
+    } else if constexpr (atoms_in_M_dim == 8 && tile_K >= 32) {
+      return XE_2D_U16x8x32_LD_N{};
+    } else {
+      return XE_2D_U16x16x16_LD_N{};
+    }
+  } else {
+    return XE_2D_U16x16x16_LD_T{};
+  }
+}
+
+// Generic way to pick a copy atom for B
+// If tuning for a specific case, create a SubgroupTilingMap specialization
+template <typename LayoutB, class TileShape_MNK, int sgs_N>
+inline auto pick_load_atom_for_B() {
+  if constexpr (cute::is_same_v<LayoutB, cutlass::layout::RowMajor>) {
+    constexpr int tile_N = get<1>(TileShape_MNK{});
+    constexpr int tile_K = get<2>(TileShape_MNK{});
+    constexpr int atoms_in_N_dim = tile_N / sgs_N;
+    if constexpr (atoms_in_N_dim >= 32 && tile_K >= 32) {
+      return XE_2D_U16x32x32_LD_V{};
+    } else {
+      return XE_2D_U16x16x16_LD_V{};
+    }
+  } else {
+    return XE_2D_U16x16x16_LD_T{};
+  }
+}
+
+// Lookup table for subgroup layout and copy instrinsic
+// This is the default case
+template <typename TileShape, typename LayoutA, typename LayoutB>
+struct SubgroupTilingMap {
+  private:
+      static constexpr auto tile_M = get<0>(TileShape{});
+      static constexpr auto tile_N = get<1>(TileShape{});
+      static constexpr bool tall_and_skinny = tile_M >= tile_N;
+      static constexpr int sgs_total = 32;
+      static constexpr int atom_N = 32; // size of the copy atom in N
+      static_assert(tile_N >= atom_N, "Tile N dim must be greater than or equal to 32");
+
+  public:
+      using sgs_M = Int<calculate_sgs_in_M<LayoutA, TileShape>()>;
+      using sgs_N = Int<std::min(tile_N/atom_N, sgs_total/sgs_M::value)>;
+      using GmemTiledCopyA = decltype(pick_load_atom_for_A<LayoutA, TileShape, sgs_M{}>());
+      using GmemTiledCopyB = decltype(pick_load_atom_for_B<LayoutB, TileShape, sgs_N{}>());
+};
+
+template <>
+struct SubgroupTilingMap<Shape<_256,_256,_32>, cutlass::layout::RowMajor, cutlass::layout::RowMajor> {
```

Comment:
Do you need those specializations just to match what was previously in benchmarks?

Reply:
The benchmarks should be matched by the default case